### PR TITLE
✨ Read configuration from child packages

### PR DESCRIPTION
**Pull request by muchzill4**
### Description

Output configuration variables exposed by dependencies when calling `mbed-tools config` sub command.

Note, build will fail as it requires releases of both `mbed-devices` and `mbed-targets`.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_tools/_internal/config_cli.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from mbed_devices.mbed_tools import config_variables as mbed_devices_config_variables
+
+
+@click.command()
+def cli() -> None:
+    """Prints information about configuration variables of all child mbed packages."""
+    click.echo(_build_output(_gather_configuration_variables()))
+
+
+def _gather_configuration_variables() -> List[pdoc.Variable]:
```
Comment (rwalton-arm) [inline mbed_tools/_internal/config_cli.py]: Any reason we don't just pass `mbed_devices_config_variables` directly to `_build_output`? I don't think this single line function buys us much when we only have a single caller?

Comment (muchzill4) [reply mbed_tools/_internal/config_cli.py]: You're right, that's silly.

Comment (muchzill4) [reply mbed_tools/_internal/config_cli.py]: https://github.com/ARMmbed/mbed-tools/pull/7/commits/828bd2539eb02dfd4dcdda15fe7cc79c2bddb7aa

File: setup.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
     description="Command line interface for Mbed OS.",
     keywords="Arm Mbed OS MbedOS cli command line tools",
     include_package_data=True,
-    install_requires=["python-dotenv", "Click==7.0", "mbed-devices", "mbed_tools_lib"],
+    install_requires=["python-dotenv", "Click==7.0", "mbed-devices", "pdoc3"],
```
Comment (rwalton-arm) [inline setup.py]: We do need `mbed-tools-lib` for the `log_exception` function.

Comment (muchzill4) [reply setup.py]: https://github.com/ARMmbed/mbed-tools/pull/7/commits/470c3143173b2c6332bff8fa0b4f20c73ced164c

**Pull request merged**